include/bbp/sonata/population.h

```diff
@@ -194,7 +196,12 @@ template <typename Population>
 class SONATA_API PopulationStorage
 {
   public:
-    PopulationStorage(const std::string& h5FilePath, const std::string& csvFilePath = "");
+    PopulationStorage(const std::string& h5FilePath);
```

Comment:
would having defaults for `csvFilePath` and `hdf5_reader` make this simpler?  I think it would simplify the `binds` as well, no?  I'm probably missing something.

Reply:
Do we allow:
```
PopulationStorage("foo.h5", hdf5_reader);
```

If yes, we'll need at least:
```
    PopulationStorage(const std::string& h5FilePath, const std::string& csvFilePath = "", const Hdf5Reader& = Hdf5Reader());
    PopulationStorage(const std::string& h5FilePath, const Hdf5Reader& hdf5_reader);
```
I don't see how it'll help with the bindings.

I think there's a tiny difference that if the default is in the signature, it updates when recompiling the code using libsonata. When putting it inside the body of the function, the default should update when simply loading to a different version of libsonata.

Honestly, it's just the version I prefer if there's more then one overload.

Reply:
> Honestly, it's just the version I prefer if there's more then one overload.

Ok, sounds good.